rest/api_test.go

```diff
@@ -2857,3 +2857,37 @@ func TestAllDbs(t *testing.T) {
 	RequireStatus(t, resp, http.StatusOK)
 	require.Equal(t, fmt.Sprintf(`[{"db_name":"%s","bucket":"%s","state":"Online"}]`, rt.GetDatabase().Name, rt.GetDatabase().Bucket.GetName()), resp.Body.String())
 }
+
+// TestBufferFlush will test for http.ResponseWriter implements Flusher interface
+func TestBufferFlush(t *testing.T) {
+	rt := NewRestTester(t, &RestTesterConfig{
```

Comment:
I think we can shorten this test, because we don't necessarily care about having any documents. I also think we don't have to test with a continuous changes feed.

What we do want to test is that all our response writers support flush. I think I'd be inclined to write a different type of test like https://github.com/couchbase/sync_gateway/blob/88edb9a72a24e4d1fd00e4ec844d2e2d034a58b4/rest/counted_response_writer_test.go#L132 using the same testcases and `getResponseWriter` but have the `httptest.NewRecorder` returned by `getResponseWriter`.